config.go

```diff
@@ -119,6 +136,33 @@ func init() {
 	}
 }
 
+func parseMatchList(text string) ([]*dto.MetricFamily, error) {
+	matchFormat := fmt.Sprintf(`match: %s`, text)
+	matchRules := make(map[string][]string)
+	err := yaml.Unmarshal([]byte(matchFormat), matchRules)
+	if err != nil {
+		return nil, err
+	}
+	var metricsList []string
+	for _, v := range matchRules["match"] {
+		metricsList = append(metricsList, fmt.Sprintf("%s 0\n", v))
+	}
+
+	metricsText := strings.Join(metricsList, "")
+
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(strings.NewReader(metricsText))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't parse match rules: %s", err)
+	}
+
+	var ret []*dto.MetricFamily
+	for _, mf := range metricFamilies {
```

Comment:
https://pkg.go.dev/github.com/prometheus/common/expfmt#TextParser.TextToMetricFamilies

`TextToMetricFamilies` returns `map[string]*dto.MetricFamily`, can be assigned directly without casting into an slice (parseMatchList) and the `for` loop.

```go

var (

        // ...

        match map[string]*dto.MetricFamily

        // ...

)

func init() {

        // ...

        if value := os.Getenv("MATCH"); value != "" {
		matchList, err := parseMatchList(value)
		if err != nil {
			logrus.WithError(err).Fatalln("couldn't parse the match rules")
		}
                match = matchList
	}

        // ...

}

func parseMatchList(text string) (map[string]*dto.MetricFamily, error) {

        // ...

        metricFamilies, err := parser.TextToMetricFamilies(strings.NewReader(metricsText))
        if err != nil {
                return nil, fmt.Errorf("couldn't parse match rules: %s", err)
        }

        return metricFamilies, nil

}

func filter(name string, labels map[string]string) bool {

        // this is ok, len() of nil maps return 0
        if len(match) == 0 {
                return true
        }

        // ...

}
```

Reply:
OK, I have modified it according to your suggestion. Thank you.